client/navBar.jsx

```diff
@@ -1,11 +1,22 @@
 import React from 'react';
+import HotelDetails from './hotelDetails'
+import Home from './pages/home'
+import Search from './search'
+
 export default class NavBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchIconClicked: false
+    };
 
+  }
   render() {
+    if (this.state.searchIconClicked) return <Search />
```

Comment:
The Navbar should generally just contain links. Your App should decide what screen to show.